### DISCUSSION TITLE
String Undefined Error

### DISCUSSION THREAD

**Discussion by dizzy222** [hashlips-lab/nft-erc721-collection#350] [Q&A]
I am getting this error when trying to deploy the smart contract via truffle. Any ideas?

hardhat.config.ts:128:7 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

128       rinkeby: process.env.BLOCK_EXPLORER_API_KEY,
          ~~~~~~~
hardhat.config.ts:129:7 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

129       mainnet: process.env.BLOCK_EXPLORER_API_KEY,
          ~~~~~~~
hardhat.config.ts:132:7 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

132       polygon: process.env.BLOCK_EXPLORER_API_KEY,
          ~~~~~~~
hardhat.config.ts:133:7 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

133       polygonMumbai: process.env.BLOCK_EXPLORER_API_KEY,
          ~~~~~~~~~~~~~

error Command failed with exit code 1.

**Comment by liarco** [comment]
Hi @dizzy222,
I'm unable to reproduce this error, unless I run a `yarn upgrade`. If you upgrade the dependencies the project may brake because it has been written and tested with lower versions.

⚠️ Remember: if you make changes to the code (especially upgrading the dependencies) it's your responsibility to make sure that everything works as intended and that there are no incompatibilities with the new dependency tree (e.g. the contract supports `v3.x` of ERC721A, compatibility with `v4.x` has never been tested and we talked about it in [this video](https://www.youtube.com/watch?v=04vbjEhu4Fk)) ⚠️

That said, this specific error can be fixed by simply changing your hardhat configuration like this:
```ts
// ...
  etherscan: {
    apiKey: {
      // Ethereum
      rinkeby: process.env.BLOCK_EXPLORER_API_KEY ?? '',
      mainnet: process.env.BLOCK_EXPLORER_API_KEY ?? '',

      // Polygon
      polygon: process.env.BLOCK_EXPLORER_API_KEY ?? '',
      polygonMumbai: process.env.BLOCK_EXPLORER_API_KEY ?? '',
    },
  },
// ...
```
